rfd/xxxx-config-resource-docs.md

```diff
@@ -0,0 +1,568 @@
+---
+authors: Paul Gottschling (paul.gottschling@goteleport.com)
+state: draft
+title: NNN - Automatically Generate the Configuration Resource Reference
+---
+
+## Required Approvers
+
+- Engineering: @codingllama
```

Comment:
Add a 2nd approver for another pair of eyes? Maybe @zmb3, since he's already reviewed.